File: projects/android/Sample/app/src/main/java/com/ezored/sample/ui/viewpager/CustomViewPager.java
package com.ezored.sample.ui.viewpager;

import android.content.Context;
import android.support.v4.view.ViewPager;
import android.util.AttributeSet;
import android.view.MotionEvent;

public class CustomViewPager extends ViewPager {

    private Boolean disable = false;

    public CustomViewPager(Context context) {
        super(context);
    }

    public CustomViewPager(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    public boolean onInterceptTouchEvent(MotionEvent event) {
        return disable ? false : super.onInterceptTouchEvent(event);
    }

    @Override
    public boolean onTouchEvent(MotionEvent event) {
        return disable ? false : super.onTouchEvent(event);
    }

    public void disableScroll(Boolean disable) {
        //When disable = true not work the scroll and when disble = false work the scroll
        this.disable = disable;
    }

}

File: projects/android/Sample/app/src/main/java/com/ezored/sample/ui/viewpager/CustomViewPager.kt
package com.ezored.sample.ui.viewpager

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import androidx.viewpager.widget.ViewPager

class CustomViewPager : ViewPager {

    private var disable: Boolean = false

    constructor(context: Context) : super(context) {
        // ignore
    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        // ignore
    }

    override fun onInterceptTouchEvent(event: MotionEvent): Boolean {
        return if (disable) false else super.onInterceptTouchEvent(event)
    }

    override fun onTouchEvent(event: MotionEvent): Boolean {
        return if (disable) false else super.onTouchEvent(event)
    }

    fun disableScroll(disable: Boolean) {
        // When disable = true not work the scroll and when disble = false work the scroll
        this.disable = disable
    }
}
